test/models/LocationOfGoodsIdentificationSpec.scala

```diff
@@ -17,107 +17,113 @@
 package models
 
 import base.SpecBase
+import generators.Generators
 import models.LocationOfGoodsIdentification._
 import org.scalacheck.Arbitrary.arbitrary
 import org.scalacheck.Gen
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 import pages.locationOfGoods.LocationTypePage
 import play.api.libs.json.{JsError, JsString, Json}
+import services.LocationOfGoodsIdentificationTypeService
+import config.Constants._
 
-class LocationOfGoodsIdentificationSpec extends SpecBase with ScalaCheckPropertyChecks {
+class LocationOfGoodsIdentificationSpec extends SpecBase with ScalaCheckPropertyChecks with Generators {
+
+  private val allValues = goodsIdentificationValues
 
   "LocationOfGoodsIdentification" - {
 
     "must deserialise valid values" in {
 
-      val gen = Gen.oneOf(LocationOfGoodsIdentification.values)
-
-      forAll(gen) {
-        locationOfGoodsIdentification =>
-          JsString(locationOfGoodsIdentification.toString).validate[LocationOfGoodsIdentification].asOpt.value mustEqual locationOfGoodsIdentification
+      forAll(Gen.alphaNumStr, Gen.alphaNumStr) {
+        (qualifier, description) =>
+          val locationOfGoodsIdentification = LocationOfGoodsIdentification(qualifier, description)
+          Json
+            .parse(s"""
+                 |{
+                 |  "qualifier": "$qualifier",
+                 |  "description": "$description"
+                 |}
+                 |""".stripMargin)
+            .as[LocationOfGoodsIdentification] mustBe locationOfGoodsIdentification
       }
     }
 
     "must fail to deserialise invalid values" in {
 
-      val gen = arbitrary[String] suchThat (!LocationOfGoodsIdentification.values.map(_.toString).contains(_))
+      val gen = arbitrary[String] suchThat (!Seq(LocationOfGoodsIdentification(DesignatedLocation, "Designated location")).map(_.toString).contains(_))
 
       forAll(gen) {
         invalidValue =>
-          JsString(invalidValue).validate[LocationOfGoodsIdentification] mustEqual JsError("error.invalid")
+          JsString(invalidValue).validate[LocationOfGoodsIdentification] mustEqual JsError("error.expected.jsobject")
       }
     }
 
     "must serialise" in {
 
-      val gen = Gen.oneOf(LocationOfGoodsIdentification.values)
-
-      forAll(gen) {
-        locationOfGoodsIdentification =>
-          Json.toJson(locationOfGoodsIdentification) mustEqual JsString(locationOfGoodsIdentification.toString)
+      forAll(Gen.alphaNumStr, Gen.alphaNumStr) {
+        (qualifier, description) =>
+          val locationOfGoodsIdentification = LocationOfGoodsIdentification(qualifier, description)
+          Json.toJson(locationOfGoodsIdentification) mustBe Json.parse(s"""
+               |{
+               |  "qualifier": "$qualifier",
+               |  "description": "$description"
+               |}
+               |""".stripMargin)
       }
     }
 
     "valuesU" - {
```

Comment:
Looks like all these valuesU tests can be removed. This is testing the `LocationOfGoodsIdentificationTypeService`